File: chess_java/src/chess/validators/KnightMoveValidator.java
package chess.validators;

import chess.Board;
import chess.Spot;
import chess.utils.MessageHandler;

public class KnightMoveValidator extends PieceMoveValidator {


    @Override
    public boolean validate(Board board, Spot start, Spot end) {
        int x = Math.abs(start.getX() - end.getX());
        int y = Math.abs(start.getY() - end.getY());
        return x * y == 2;
    }

    @Override
    protected String getNotValidMessage() {
        return MessageHandler.getHellNoMessage();
    }
}

File: chess_cpp/validators/KnightMoveValidator.cpp
//
// Created by Hrach Khachatryan on 12/15/19.
//

#include "KnightMoveValidator.h"
#include "../utils/MessageHandler.h"

bool KnightMoveValidator::validate(Board *board, Spot *start, Spot *end) {
    int x = abs(start->getX() - end->getX());
    int y = abs(start->getY() - end->getY());
    return x * y == 2;
}

std::string KnightMoveValidator::getNotValidMessage() {
    return MessageHandler::getHellNoMessage();
}
